test/cmocka/src/math/fft/fft.c

```diff
@@ -0,0 +1,1252 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+//
+// Author: Keyon Jie <yang.jie@linux.intel.com>
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <math.h>
+#include <cmocka.h>
+#include <stdbool.h>
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/drivers/ipc.h>
+#include <sof/math/fft.h>
+
+/* SNR DB threshold in Q8.24 */
+#define FFT_DB_TH	Q_CONVERT_FLOAT(40.0, 24)
+
+#define SINE_HZ	1000
+
+/* sine wave samples, freq = 1000Hz */
+int input_samples[FFT_SIZE_MAX] = {
```

Comment:
Can this be in a separate header file?

Reply:
> Can this be in a separate header file?

sure, maybe test/cmocka/src/math/fft/sine_1k_wav.h?